src/init/common.h

```diff
@@ -8,13 +8,17 @@
 #ifndef BITCOIN_INIT_COMMON_H
 #define BITCOIN_INIT_COMMON_H
 
+#include <util/translation.h>
+
+#include <optional>
+
 class ArgsManager;
 
 namespace init {
 void AddLoggingArgs(ArgsManager& args);
 void SetLoggingOptions(const ArgsManager& args);
-void SetLoggingCategories(const ArgsManager& args);
-void SetLoggingLevel(const ArgsManager& args);
+std::optional<bilingual_str> SetLoggingCategories(const ArgsManager& args);
+std::optional<bilingual_str> SetLoggingLevel(const ArgsManager& args);
```

Comment:
nit: document those as the return value may not be so obvious to everybody: "if a non-empty optional is returned then an error occurred and it contains the error message".

Reply:
> nit: document those as the return value may not be so obvious to everybody: "if a non-empty optional is returned then an error occurred and it contains the error message".

It's good to document, but this convention does seem to be increasingly used. I have a PR to replace `std::optional<bilingual_str>` with `util::Result` various places (74655e5870888e0165c62fec75000fe04f062147, #25977)

Reply:
Using `util::Result` here would be self-documenting.